src/middlewared/middlewared/plugins/jail.py

```diff
@@ -631,6 +664,55 @@ def list_resource(self, resource, remote, want_cache, branch):
 
         return resource_list
 
+    @periodic(interval=86400)
+    @private
+    @accepts(Str('branch', null=True, default=None))
+    @job(lock='retrieve_plugin_versions')
+    def retrieve_plugin_versions(self, job, branch=None):
+        branch = branch or self.get_version()
+        ioc_plugin = IOCPlugin(branch=branch)
+        if self.get_activated_pool():
+            plugins = ioc_plugin.fetch_plugin_versions()
+        else:
+            # iocage hasn't been activated yet, we should use requests to retrieve plugin json files
+            session = requests.Session()
+            url = 'https://raw.githubusercontent.com/freenas/iocage-ix-plugins/{branch}/{file_name}'
```

Comment:
This is the same problem as before, one request for each plugin.

Why is this not in iocage?